include/deal.II/base/subscriptor.h

```diff
@@ -104,21 +104,44 @@ class Subscriptor
 
   /**
    * Subscribes a user of the object by storing the pointer @p validity. The
-   * subscriber may be identified by text supplied as @p identifier.
+   * subscriber may be identified by text supplied as @p identifier. The latter
+   * variable must be not be a temporary and its type must decay to
+   * const char *. In particular, calling this function with a rvalue reference
+   * is not allowed.
+   */
+  template <typename ConstCharStar = const char *>
+  typename std::enable_if<
+    std::is_same<ConstCharStar, const char *>::value>::type
+  subscribe(std::atomic<bool> *const validity,
+            ConstCharStar            identifier = nullptr) const;
```

Comment:
Why is it necessary to do the `std::is_same` thing? Can't you just use `const char *` as argument type?

Reply:
IMHO, the error messages are then pretty confusing. Basically, the function call is ambiguous for rvalue references.
https://godbolt.org/z/t2XXVy

Reply:
Ah, that's too bad. Thanks for checking, though!